cmd/root.go

```diff
@@ -80,6 +80,32 @@ It would also download chapters 10 to 20 of Black Clover from mangadex.org in Sp
 			os.Exit(1)
 		}
 
+		chapters = chapters.SortByNumber()
+
+		rangesIndex := 1
+		// ranges argument is not provided
+		if rangesIndex >= len(args) {
+			lastChapter := chapters[len(chapters)-1].GetNumber()
+			prompt := promptui.Prompt{
+				Label:     fmt.Sprintf("Do you want to download all %g chapters", lastChapter),
+				IsConfirm: true,
+			}
+
+			_, err := prompt.Run()
+
+			if err != nil {
+				color.Yellow("No ranges specified")
+				os.Exit(0)
+			}
+
+			args = append(args, fmt.Sprintf("1-%g", lastChapter))
```

Comment:
rather than using a string, let's avoid parsing the ranges. Instead, define the `rngs`:

~~~go
rngs = []Range{Range{Begin: 0, End: lastChapter}}
~~~

Note `rngs` would need to be set before the condition, that's why I'm not using `:=` here.

Reply:
Changed in the [925d587](https://github.com/elboletaire/manga-downloader/pull/7/commits/925d5872f4918e411fe9ae01f0890527fc14e721) commit.